heft-plugins/heft-rspack-plugin/src/RspackPlugin.ts

```diff
@@ -0,0 +1,489 @@
+        let devServerOptions: TRspackDevServer.Configuration;
+        if (Array.isArray(rspackConfiguration)) {
+          const filteredDevServerOptions: TRspackDevServer.Configuration[] = rspackConfiguration
+            .map((configuration) => configuration.devServer)
+            .filter((devServer): devServer is TRspackDevServer.Configuration => !!devServer);
+          if (filteredDevServerOptions.length > 1) {
+            taskSession.logger.emitWarning(
+              new Error(`Detected multiple rspack devServer configurations, using the first one.`)
+            );
+          }
+          devServerOptions = { ...defaultDevServerOptions, ...filteredDevServerOptions[0] };
+        } else {
+          devServerOptions = { ...defaultDevServerOptions, ...rspackConfiguration.devServer };
+        }
+
+        // Add the certificate and key to the devServerOptions if these fields don't already have values
+        if (!devServerOptions.server) {
+          const certificateManager: CertificateManager = new CertificateManager();
+          const certificate: ICertificate = await certificateManager.ensureCertificateAsync(
+            true,
+            taskSession.logger.terminal
+          );
+
+          // Update the web socket URL to use the hostname provided by the certificate
+          const clientConfiguration: TRspackDevServer.Configuration['client'] = devServerOptions.client;
+          const hostname: string | undefined = certificate.subjectAltNames?.[0];
+          if (hostname && typeof clientConfiguration === 'object') {
+            const { webSocketURL } = clientConfiguration;
+            if (typeof webSocketURL === 'object') {
+              clientConfiguration.webSocketURL = {
+                ...webSocketURL,
+                hostname
+              };
+            }
+          }
+
+          devServerOptions = {
+            ...devServerOptions,
+            server: {
+              type: 'https',
+              options: {
+                minVersion: 'TLSv1.3',
+                key: certificate.pemKey,
+                cert: certificate.pemCertificate,
+                ca: certificate.pemCaCertificate
+              }
+            }
+          };
+        }
+
+        // Since the webpack-dev-server does not return infrastructure errors via a callback like
+        // compiler.watch(...), we will need to intercept them and log them ourselves.
+        // note: @rspack/dev-server extends webpack-dev-server and also has this behavior
+        compiler.hooks.infrastructureLog.tap(
+          PLUGIN_NAME,
+          (name: string, type: string, args: unknown[] | undefined) => {
+            if (name === WEBPACK_DEV_MIDDLEWARE_PACKAGE_NAME && type === 'error') {
+              const error: Error | undefined = args?.[0] as Error | undefined;
+              if (error) {
+                taskSession.logger.emitError(error);
+              }
+            }
+            return true;
```

Comment:
There's a bugfix out to remove this from the webpack plugin; probably we should only `return true` inside of `if (error)`